language/diem-vm/src/diem_vm.rs

```diff
@@ -221,21 +224,49 @@ impl DiemVMImpl {
         let txn_expiration_timestamp_secs = txn_data.expiration_timestamp_secs();
         let chain_id = txn_data.chain_id();
         let mut gas_status = GasStatus::new_unmetered();
+        let secondary_public_key_hashes: Vec<MoveValue> = txn_data
+            .secondary_authentication_key_preimages
+            .iter()
+            .map(|preimage| {
+                MoveValue::vector_u8(HashValue::sha3_256_of(&preimage.to_vec()).to_vec())
+            })
+            .collect();
+        let args = if self.get_diem_version()? >= DIEM_VERSION_2 && txn_data.is_multi_agent() {
```

Comment:
I'm still not sure about this, but I'm probably misunderstanding something. From what I understand reading this, this is saying if the diem version is 2 or greater and the transaction is multi-agent it will use a multi-agent feature. But that doesn't seem right since these features should only become active at version 3 and greater?

Reply:
@emmazzz My two concerns on versioning are here and the code just below this as well (that similarly checks a `>= DIEM_VERSION_2`)

Reply:
@tzakian you are right. It should be DIEM_VERSION_3 instead. I have updated it here and below this. Thanks for pointing this out!